int64 timestamp
char gear